src/EnergyPlus/ConvectionConstants.hh

```diff
@@ -184,66 +444,112 @@ enum class InConvWinLoc
 };
 
 // Parameters for adaptive convection algorithm's classification of inside face of surfaces
-enum class InConvClass
+enum class IntConvClass
 {
     Invalid = -1,
-    A1_VertWalls,          // flow regime A1, vertical walls
-    A1_StableHoriz,        // flow regime A1
-    A1_UnstableHoriz,      // flow regime A1
-    A1_HeatedFloor,        // flow regime A1
-    A1_ChilledCeil,        // flow regime A1
-    A1_StableTilted,       // flow regime A1
-    A1_UnstableTilted,     // flow regime A1
-    A1_Windows,            // flow regime A1
-    A2_VertWallsNonHeated, // flow regime A2
-    A2_HeatedVerticalWall, // flow regime A2
-    A2_StableHoriz,        // flow regime A2
-    A2_UnstableHoriz,      // flow regime A2
-    A2_StableTilted,       // flow regime A2
-    A2_UnstableTilted,     // flow regime A2
-    A2_Windows,            // flow regime A2
-    A3_VertWalls,          // flow regime A3
-    A3_StableHoriz,        // flow regime A3
-    A3_UnstableHoriz,      // flow regime A3
-    A3_StableTilted,       // flow regime A3
-    A3_UnstableTilted,     // flow regime A3
-    A3_Windows,            // flow regime A3
-    B_VertWalls,           // flow regime B
-    B_VertWallsNearHeat,   // flow regime B
-    B_StableHoriz,         // flow regime B
-    B_UnstableHoriz,       // flow regime B
-    B_StableTilted,        // flow regime B
-    B_UnstableTilted,      // flow regime B
-    B_Windows,             // flow regime B
-    C_Walls,               // flow regime C
-    C_Ceiling,             // flow regime C
-    C_Floor,               // flow regime C
-    C_Windows,             // flow regime C
-    D_Walls,               // flow regime D
-    D_StableHoriz,         // flow regime D
-    D_UnstableHoriz,       // flow regime D
-    D_StableTilted,        // flow regime D
-    D_UnstableTilted,      // flow regime D
-    D_Windows,             // flow regime D
-    E_AssistFlowWalls,     // flow regime E
-    E_OpposFlowWalls,      // flow regime E
-    E_StableFloor,         // flow regime E
-    E_UnstableFloor,       // flow regime E
-    E_StableCeiling,       // flow regime E
-    E_UnstableCeiling,     // flow regime E
-    E_Windows,             // flow regime E
+    // SimpleBuoy goes first in the IDF objects, so has to go first here too, A3 or not.
```

Comment:
Here is something cute. In the IDF, the `A3` regime is listed first. And thus it must be listed first here also.